modules/py_image.c

```diff
@@ -1269,7 +1269,7 @@ static MP_DEFINE_CONST_FUN_OBJ_KW(py_image_save_obj, 2, py_image_save);
 
 static mp_obj_t py_image_flush(mp_obj_t img_obj) {
     framebuffer_t *fb = framebuffer_get(0);
-    framebuffer_update_jpeg_buffer(fb);
+    framebuffer_update_jpeg_buffer(fb, py_image_cobj(img_obj));
```

Comment:
It was meant to update the FB after drawing to it, since all images originally were FB pointers. I think the cleaner way to fix this would be to refactor `framebuffer_update_jpeg_buffer` to accept an image, since that's what it really expects. You could probably reuse the image pointer passed to every `snapshot`.

Reply:
I was actually going to do that initially, but, it calls a bunch of functions internally that need the FB pointer. However, none of the functions it passes the FB pointer to actually use the pointer... so, I can drop the pointer from those functions and almost everything in framebuffer_update_jpeg_buffer. 

But, it does check `fb->streaming_enabled` internally. How should I handle this? This expands the scope of the refactor.

Reply:
I thought about how to refactor this to get rid of the fb_streaming_enabled issue. Will send this in another PR.

Reply:
Let's see it, but on second look it does expect a framebuffer, so maybe just copy the image to be flushed to fb if it's not already the fb ? And let's remove the `src_image` confusing stuff, it's not needed.

Reply:
> so maybe just copy the image to be flushed to fb if it's not already the FB

Not possible, the FB in event mode is like 1024x4 pixels. Have to decouple.

Reply:
Not following? When calling `img.flush()` if the image != FB you could just memcpy it to fb, before calling `update_jpeg_buffer`.

Reply:
In event mode the fb is like 1024x4 bytes pixels which represents 1024 events (4-bytes per event). The generated image is 320x320 pixels. So, I can't even put the image in the frame buffer. I need to directly flush the image into the jpeg buffer.

Reply:
I see the problem. Okay, we could refactor `framebuffer_update_jpeg_buffer` to call a new function that accepts `image_t`, which could also be called by flush. `framebuffer_update_jpeg_buffer` can still do all the checks, like `streaming_enabled` etc.. before calling the new function. That mode that prints the image to serial, how useful is it? We could also remove it.

Reply:
> I'm thinking dropping it and just printing a message that the frame buffer can't be displayed because the image is too large

Yes we should drop it. As for the message, and text dropping, I don't like printing a lot of messages. Can we draw the error on the frame buffer instead?

> As for the refactor, I was actually going to completely remove the streaming_enabled and the logic in the omv module that controls it as you can just pass update=False to sensor now. So, it's not needed at all anymore.

Okay, I never really liked it, seems confusing to have to flags that control streaming.

Reply:
> Yes we should drop it. As for the message, and text dropping, I don't like printing a lot of messages. Can we draw the error on the frame buffer instead?

Good idea. I can send an escape sequence and have OpenMV IDE decode it.

Reply:
https://github.com/openmv/openmv/pull/2755 - Will rebase after PR is merged.